### PR TITLE
Add table cities and streets and update orders table

### PR DESCRIPTION
**Pull request by MM3103**

### CODE REVIEW

File: src/main/java/my/pr/controller/CityController.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        @ApiResponse(responseCode = "401", description = "Access error"),
+        @ApiResponse(responseCode = "403", description = "Access error"),
+        @ApiResponse(responseCode = "400", description = "Missing or invalid request body"),
+        @ApiResponse(responseCode = "500", description = "Internal error")})
+public class CityController {
+
+    @Autowired
+    CityService service;
+
+    @GetMapping("/city/getAllCity")
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/controller/CityController.java:30]: /city -> place it into @RestController as parameter

File: src/main/java/my/pr/controller/StreetController.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        @ApiResponse(responseCode = "401", description = "Access error"),
+        @ApiResponse(responseCode = "403", description = "Access error"),
+        @ApiResponse(responseCode = "400", description = "Missing or invalid request body"),
+        @ApiResponse(responseCode = "500", description = "Internal error")})
+public class StreetController {
+
+    @Autowired
+    StreetService service;
+
+    @GetMapping("/street/getAllStreet")
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/controller/StreetController.java:30]: /street -> place it into @RestController as parameter

File: src/main/java/my/pr/controller/StreetController.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    @Operation(summary = "Get all street")
+    public List<Street> getAllStreets() {
+        return service.getAllStreets();
+    }
+
+    @GetMapping("/street/getStreet/{id}")
+    @Operation(summary = "Get street by id")
+    public ResponseEntity<Street> getStreet(@PathVariable(value = "id") Long id) throws EntityNotFoundException {
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/controller/StreetController.java:38]: getStreet -> getStreetById

File: src/main/java/my/pr/controller/StreetController.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        return ResponseEntity.ok().body(service.getStreet(id));
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/controller/StreetController.java:39]: getStreet -> getStreetById

File: src/main/java/my/pr/controller/StreetController.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @PostMapping("/street/add")
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/controller/StreetController.java:42]: add - not necessary

File: src/main/java/my/pr/controller/StreetController.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    @Operation(summary = "Add new street")
+    public Street addStreet(@RequestBody Street newStreet) {
+        return service.addCity(newStreet);
+    }
+
+    @DeleteMapping("/street/delete/{id}")
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/controller/StreetController.java:48]: /street/delete/{id} -> /street/{id}

File: src/main/java/my/pr/controller/StreetController.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    @Operation(summary = "Delete street by id")
+    public void deleteStreet(@PathVariable(value = "id") Long id) throws EntityNotFoundException {
+        service.deleteStreet(id);
+    }
+
+    @PatchMapping("/street/update/{id}")
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/controller/StreetController.java:54]: /street/update/{id} -> /street/{id}

File: src/main/java/my/pr/controller/StreetController.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    @Autowired
+    StreetService service;
+
+    @GetMapping("/street/getAllStreet")
+    @Operation(summary = "Get all street")
+    public List<Street> getAllStreets() {
+        return service.getAllStreets();
+    }
+
+    @GetMapping("/street/getStreet/{id}")
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/controller/StreetController.java:36]: /street/getStreet/{id} -> /street/{id}

File: src/main/java/my/pr/model/Order.java [hunk 9]
```diff
@@ -38,19 +39,36 @@ public Order() {
...
+
+    @NotNull
+    @Column(name = "house", nullable = false)
+    private Integer house;
+
+    @Column(name = "flat", nullable = false)
+    private Integer flat;
+
+    @Column(name = "installation", nullable = false)
+    private Boolean installation;
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/model/Order.java:56]: selfInstallation

File: src/main/java/my/pr/repository/CityRepository.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface CityRepository  extends JpaRepository<City,Long> {
+
+    @Query(value = "select * from cities where name = :name ", nativeQuery = true)
+    City findByName(@Param("name") String name);
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/repository/CityRepository.java:13]: i think, it should work without @Query

File: src/main/java/my/pr/security/WebSecurityConfig.java [hunk 11]
```diff
@@ -41,8 +41,8 @@ protected void configure(HttpSecurity http) throws Exception {
                 .csrf().disable()
                 .cors().and()
                 .authorizeRequests()
-                .antMatchers("/order", "/order/getUserOrders").hasAnyRole("ADMIN", "USER")
-                .antMatchers("/order/acceptOrder/**", "/order/rejectOrder/**", "/order/getAll").hasRole("ADMIN")
+                .antMatchers("/order", "/order/getUserOrders","/city/getAllCity","/city/getCity/*","/street/getAllStreet","/street/getStreet/*").hasAnyRole("ADMIN", "USER")
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/security/WebSecurityConfig.java:44]: try to change it to /city/*

File: src/main/java/my/pr/service/CityService.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    public List<City> getAllCities() {
+        return repository.findAll();
+    }
+
+    public City getCity(Long id) throws EntityNotFoundException {
+        return repository.findById(id).orElseThrow(() -> new EntityNotFoundException("City not found for id: " + id));
+    }
+
+    public Long getIdByName(String name) throws EntityNotFoundException {
+/*        City city = repository.findById(name).orElseThrow(() -> new EntityNotFoundException("City not found for id: " + name));*/
```
Comment (AntonPozdeev) [inline src/main/java/my/pr/service/CityService.java:27]: delete it

File: src/main/java/my/pr/controller/CityController.java (hunk 1 above)
Comment (MM3103) [reply src/main/java/my/pr/controller/CityController.java]: fixed

File: src/main/java/my/pr/controller/StreetController.java (hunk 2 above)
Comment (MM3103) [reply src/main/java/my/pr/controller/StreetController.java]: fixed

File: src/main/java/my/pr/controller/StreetController.java (hunk 3 above)
Comment (MM3103) [reply src/main/java/my/pr/controller/StreetController.java]: fixed

File: src/main/java/my/pr/controller/StreetController.java (hunk 4 above)
Comment (MM3103) [reply src/main/java/my/pr/controller/StreetController.java]: fixed

File: src/main/java/my/pr/model/Order.java (hunk 9 above)
Comment (MM3103) [reply src/main/java/my/pr/model/Order.java]: fixed

File: src/main/java/my/pr/service/CityService.java (hunk 12 above)
Comment (MM3103) [reply src/main/java/my/pr/service/CityService.java]: fixed

**Pull request merged**